cantools/database/dataelement.py

```diff
@@ -120,13 +129,21 @@ def choice_string_to_number(self, string: str) -> int:
 
         raise KeyError(f"Choice {string} not found in data element {self.name}.")
 
-    def _get_offset_scaling_from_list(self, val: float, segments: PiecewiseSegment) -> Tuple[float, float]:
-        for i, segment in enumerate(segments):
-            start, end = segment
-            if start <= val <= end:
-                return self._offset[i], self._scale[i]  # type: ignore
+    def _get_offset_scaling_from_list(
+        self, val: float, segments: Tuple[List[float], List[float]]
+    ) -> Tuple[float, float]:
+        i = (
+            bisect.bisect_right(segments[0], val) - 1
+        )  # lower limits: rightmost value less than or equal val
+        i2 = bisect.bisect_left(
+            segments[1], val
+        )  # higher limits: leftmost value greater than or equal val
```

Comment:
I think that this will blow up if segments overlap. for the sake of clarity -- and given the fact that piecewise linear transformations are pretty niche -- I think that it is acceptable to stay with the O(N) algorithm for now (I suppose that N is usually pretty small as well)...